lib/Model.ts

```diff
@@ -673,7 +673,7 @@ export class Model<TDocument, TInstance> {
      * @returns {Promise<number>}
      */
     // @ts-ignore We implement support for leaving out `options` but providing the `callback` in our base implementation, even though this complains
-    remove(conditions: { _id?: string; } | Conditions | any, callback?: General.Callback<number>): Promise<number>;
+    remove(conditions: { _id?: string; } | Conditions<TDocument> | any, callback?: General.Callback<number>): Promise<number>;
```

Comment:
I was getting a typescript building error in my personal project when using alpha-13 and it was complaining that `Conditions` was missing a type argument. After looking at all of the other places I think you just missed this one. And its not surprising that you missed it since you already have a `@ts-ignore` comment for the ts complaining about an improper overload,

So let me know if I am wrong in putting `<TDocument>` here.